src/test/groovy/graphql/schema/validation/TypesImplementInterfacesTest.groovy

```diff
@@ -317,4 +320,82 @@ class TypesImplementInterfacesTest extends Specification {
         goodErrorCollector.getErrors().isEmpty()
     }
 
+    def "type can declare extra optional field arguments"() {
```

Comment:
we should add also an SDL based test, because we are currently not always consistent.

Reply:
There are current tests that cover that for the SDL already in `graphql.schema.idl.SchemaTypeCheckerTest`.

- `test field arguments on object can contain additional optional arguments`

Currently, the SDL validation doesn't reproduce the error described in the ticket.